zebra-network/src/peer_set/initialize.rs

```diff
@@ -285,20 +287,23 @@ where
                 connection_tracker,
             };
 
-            let outbound_connector = outbound_connector.clone();
-            async move {
-                // Rate-limit the connection, sleeping for an interval according
-                // to its index in the list.
+            // Construct a connector future but do not drive it yet ...
+            let outbound_connector_future = outbound_connector
+                .clone()
+                .oneshot(req)
+                .map_err(move |e| (addr, e));
+
+            // ... instead, spawn a new task to handle this connector
+            tokio::spawn(async move {
+                let task = outbound_connector_future.await;
+                // Only spawn one outbound connector per `MIN_PEER_CONNECTION_INTERVAL`.
                 sleep(constants::MIN_PEER_CONNECTION_INTERVAL.saturating_mul(i as u32)).await;
-                outbound_connector
-                    .oneshot(req)
-                    .map_err(move |e| (addr, e))
-                    .await
-            }
+                task
+            })
         })
         .collect();
 
-    while let Some(handshake_result) = handshakes.next().await {
+    while let Some(Ok(handshake_result)) = handshakes.next().await {
```

Comment:
Error handling: Can we turn handshake task panics into panics in the calling task?

If we just check for `Ok`, we will stop connecting to peers if a handshake panics, and continue running Zebra. So we won't actually see the panics in the logs.

(I can do a similar fix for inbound connections.)

```suggestion
    while let Some(handshake_result) = handshakes.next().await {
        let handshake_result = handshake_result.expect("panic in initial peer handshake");
```

Reply:
Thanks, changed to this at https://github.com/ZcashFoundation/zebra/pull/3189/commits/36f708c82875c1a1ddc0858eb284fd948ee74688